src/main/java/slimeknights/tconstruct/library/tools/item/ToolCore.java

```diff
@@ -374,7 +379,80 @@ public void inventoryTick(ItemStack stack, World worldIn, Entity entityIn, int i
       }
     }
   }
+  
+  /* Right click hooks */
+  
+  @Override
+  public ActionResultType onItemUseFirst(ItemStack stack, ItemUseContext context) {
+    ToolStack tool = ToolStack.from(stack);
+    for (ModifierEntry entry : tool.getModifierList()) {
+      ActionResultType result = entry.getModifier().onItemUse(tool, entry.getLevel(), stack, context);
+      if (result != ActionResultType.PASS) {
```

Comment:
Vanilla has two variants of `PASS`, the other is `FAIL`, should probably be using `isSuccessOrConsume` instead of an equals check, or were you intending `FAIL` to prevent other modifier hooks from running?

Reply:
Nah, modifiers that want to stop without running code can just `CONSUME`.